common/factorization/par_ilu_kernels.hpp.inc

```diff
@@ -33,6 +33,203 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 namespace kernel {
 
 
+namespace detail {
+
+
+// Default implementation for the unsorted case
+template <bool IsSorted>
+struct find_helper {
```

Comment:
This `find_helper` looks pretty generic. Maybe it can be moved to a common_kernels or something ? A find function i\can be pretty useful in a lot of cases.

Reply:
I would argue that in the current form, the function is not really general-purpose enough to be useful elsewhere. It might be helpful if it returned the index of the found value (or -1 if it didn't find the value), but that might complicate the rest of the code.

Reply:
Yes, now that I see it does return a boolean.